src/request_handler_static.cpp

```diff
@@ -1,124 +1,127 @@
-//   }
+  //Request path must be absolute and not contain "..".
+  if (request_file.empty() || request_file[0] != '/'
+      || request_file.find("..") != std::string::npos) {
+    return RequestHandler::BAD_REQUEST;
+  }
 
-//   // Determine the file extension.
-//   std::size_t last_slash_pos = request_file.find_last_of("/");
-//   std::size_t last_dot_pos = request_file.find_last_of(".");
-//   std::string extension;
-//   if (last_dot_pos != std::string::npos && last_dot_pos > last_slash_pos) {
-//     extension = request_file.substr(last_dot_pos + 1);
-//   }
+  // If path ends in slash (i.e. is a directory) then add "index.html".
+  if (request_file[request_file.size() - 1] == '/') {
+    request_file += "/index.html";
+  }
 
-//   // Open the file to send back.
-//   std::string full_path = static_file_root_ + request_file;
-//   std::ifstream is(full_path.c_str(), std::ios::in | std::ios::binary);
-//   if (!is) {
-//     rep = reply::stock_reply(reply::not_found);
-//     return;
-//   }
+  // Determine the file extension.
+  std::size_t last_slash_pos = request_file.find_last_of("/");
+  std::size_t last_dot_pos = request_file.find_last_of(".");
+  std::string extension;
+  if (last_dot_pos != std::string::npos && last_dot_pos > last_slash_pos) {
+    extension = request_file.substr(last_dot_pos + 1);
+  }
 
-//   // Fill out the reply to be sent to the client.
-//   rep.status = reply::ok;
-//   char buf[512];
-//   while (is.read(buf, sizeof(buf)).gcount() > 0)
-//     rep.content.append(buf, is.gcount());
-//   rep.headers.resize(2);
-//   rep.headers[0].name = "Content-Length";
-//   rep.headers[0].value = boost::lexical_cast<std::string>(rep.content.size());
-//   rep.headers[1].name = "Content-Type";
-//   rep.headers[1].value = mime_types::extension_to_type(extension);
-// }
+  // Open the file to send back.
+  std::string full_path = root + request_file;
+  std::ifstream is(full_path.c_str(), std::ios::in | std::ios::binary);
+  if (!is) {
+    return RequestHandler::IOERROR;
+  }
 
-// bool request_handler_static::url_decode(const std::string& in, std::string& out) {
-//   out.clear();
-//   out.reserve(in.size());
-//   for (std::size_t i = 0; i < in.size(); ++i) {
-//     if (in[i] == '%') {
-//       if (i + 3 <= in.size()) {
-//         int value = 0;
-//         std::istringstream is(in.substr(i + 1, 2));
-//         if (is >> std::hex >> value) {
-//           out += static_cast<char>(value);
-//           i += 2;
-//         }
-//         else {
-//           return false;
-//         }
-//       }
-//       else {
-//         return false;
-//       }
-//     }
-//     else if (in[i] == '+') {
-//       out += ' ';
-//     }
-//     else {
-//       out += in[i];
-//     }
-//   }
-//   return true;
-// }
+  // Fill out the reply to be sent to the client.
+  response->SetStatus(Response::OK);
+  char buf[512];
+  std::string temp = "";
+  while (is.read(buf, sizeof(buf)).gcount() > 0)
```

Comment:
maybe enclosed this in brackets, just for style/consistency's sake (even though it's just one line in the loop!)